controllers/datafoundationclient.go

```diff
@@ -125,6 +136,27 @@ func (r *dataFoundationClientReconciler) reconcilePhases() (ctrl.Result, error)
 	if !r.offering.DeletionTimestamp.IsZero() {
 		return ctrl.Result{}, nil
 	} else {
+
+		// Checking for CRDs that are installed after the agent was started and add watchers for the new CRDs
+		if !r.ManagedFusionOfferingReconciler.AvailableCRDs[StorageClientCRDName] {
+			crd := apiextensionsv1.CustomResourceDefinition{}
+			crd.Name = StorageClientCRDName
+			if err := r.get(&crd); err != nil {
+				return ctrl.Result{}, err
+			} else {
+				utils.HandleMissingWatchers(StorageClientCRDName)
+			}
+		}
+		if !r.ManagedFusionOfferingReconciler.AvailableCRDs[StorageClassClaimCRDName] {
```

Comment:
Please do not access the underlaying reconciler in the offering code.
You can copy a ref to the AvailableCRDs map into the sub reconciler on init